src/test/java/za/ac/cput/factory/VehicleFactoryTest.java

```diff
@@ -0,0 +1,23 @@
+/*      VehicleFactoryTest.java
+        Test for the Vehicle Factory
+        Author: Simphiwe Kahlana (220162891)
+        Date: 22 March 2023
+        */
+package za.ac.cput.factory;
+
+import org.junit.jupiter.api.Test;
+import za.ac.cput.domain.Branch;
+import za.ac.cput.domain.Vehicle;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class VehicleFactoryTest {
+    @Test
+    public void testVehicleFactory(){
+        Vehicle vehicle = VehicleFactory.createVehicle("BMW2891", "Convertible","M326s", "Sport", "Black",
```

Comment:
Here, because the id is being generated in the factory and you removed the VehicleId parameter, you would just have the 4 arguments passed in here without the id.

```suggestion
        Vehicle vehicle = VehicleFactory.createVehicle("Convertible","M326s", "Sport", "Black",
```